### DISCUSSION TITLE
[Chapter 11] 배치 프로세싱이 너무 많은 경우 효과적인 관리 방법은?

### DISCUSSION THREAD

**Discussion by corock** [cowaine/spring-boot-for-msa#178] [Chapter 11. Spring Scheduling Tasks] [answered]
제목 그대로 하나의 서비스와 관련지어 뒤에서 돌고 있는 배치가 너무 많은 경우 자칫 혼란을 줄 수도 있을 것 같은데요.

예를 들어, 특정 서비스에서 A 라는 API 가 있고, B 배치가 있을 때, A, B 는 모두 같은 테이블의 특정 컬럼 값을 업데이트한다고 가정해 봅시다.
공교롭게 트래픽이 몰리는 시점에 A API 가 많이 호출되었고, B 배치로 인해 같은 테이블의 특정 값을 업데이트하던 도중 데드락이 발생했습니다.
API 서버는 장애로 이어졌고, 원인을 추적하기 위해 로그를 분석했으나 데드락과 관련된 로그는 B 배치 서버에 적재되어 있습니다.
심지어 이 현상을 뒤늦게 파악하고, 그에 대한 해결책으로 만약 C 라는 배치를 추가하는 방향으로 간다면? 그리고 이런 상황이 누적되어 되풀이된다면?

위와 같은 상황에 놓이게 되면 효과적으로 관리할 수 있는 방법은 어떤 것들이 있을까? 등이 궁금해 지는데요.
예를 들면, 서비스와 관련된 배치 목록을 한 눈에 봤으면 좋겠다거나, 위키 페이지로 사용 중인 배치 목록을 현행화해서 관리한다거나, ...
이렇듯 배치 프로세싱이 너무 많은 경우에 여러분이 생각하는 효과적인 관리 방법에 대한 의견이 궁금합니다.

**Comment by JoisFe** [comment, marked as answer]
책에서도 젠킨스를 이용해 스케줄링과 각종 명령어를 실행하는 예시가 있는데
@corock 님께서 언급한 배치작업이 너무 많아 관리가 어려운 경우 젠킨스를 이용해서 관리하는 예시가 있어 링크 남깁니다.
[젠킨스로 스프링 배치의 관리하기](https://jojoldu.tistory.com/489)

저희 회사 팀원 한분은 Spring Batch 애플리케이션을 컨테이너 단위로 도입한 후 `Spring Cloud Data Flow`를 도입하여 배치 애플리케이션 관리하기 위한 시도를 진행하고 있는 것 으로 압니다.
[Spring Cloud Data Flow](https://dataflow.spring.io/docs/batch-developer-guides/batch/data-flow-spring-batch/)

- 다만 참고 자료가 많이 부족하여 K사 컨퍼런스 등을 다녀오는 등 고군분투 하고 있는 것으로 압니다.
- [K 사 SCDF 사용 예시](https://tv.kakao.com/channel/3693125/cliplink/414072537)
